### DISCUSSION TITLE
Difficulty with Built-in Tool Returning a Docx File

### DISCUSSION THREAD

**Discussion by Greyth-l** [langgenius/dify#3269] [Help]
Hi there,

I’m currently working on a built-in tool designed to send a POST HTTP request to a backend service, which in turn returns a .docx file. I’ve implemented the tool as follows:

```python
response = requests.post(url, json=body)
if response.status_code == 200:
    logging.info("generate file success!")
    return [
        self.create_text_message('generate file success'),
        self.create_blob_message(blob=response.content,
                                 meta={
                                     'mime_type': 'application/vnd.openxmlformats-officedocument.wordprocessingml.document'})
    ]
else:
    logging.info("generate file failed!")
    return self.create_text_message(f'generate file failed, code is {response.status_code}')
```

The tool does generate a document, which is stored at the following path: `\api\storage\tools\bb445406-e22d-4b68-af4f8-d402a7f85afb\79ecf6cadaef45209847b38a8bf3f1b3.docx`.

However, the LLM responds with a ‘result link’ that points to `/files/tools/177629c4-8de8-4140-938b-a952d94d7d6d.docx`, which I cannot locate in my directory. Furthermore, when I attempt to access this URL, it results in a 404 error.

Could you please assist me in resolving this issue? I’m not sure why the LLM is providing an incorrect file path, and I would appreciate any guidance on how to correctly link to the generated document.

Thank you!

**Comment by Yeuoly** [comment]
hi, I'm sorry for it, but this mimetype is not supported yet, our frontend has not implemented a solution for docx, pptx, xlsx, etc.
